tests/unit-core/test-external-string.c

```diff
@@ -53,38 +65,45 @@ main (void)
   jerry_init (JERRY_INIT_EMPTY);
 
   /* Test external callback calls. */
-  jerry_value_t external_string = jerry_create_external_string ((jerry_char_t *) external_1, free_external1);
+  jerry_string_set_external_string_free_callback (external_string_free_callback_1);
+  jerry_value_t external_string = jerry_create_external_string ((jerry_char_t *) external_1, NULL);
   TEST_ASSERT (free_count == 0);
   jerry_release_value (external_string);
   TEST_ASSERT (free_count == 1);
 
+  jerry_string_set_external_string_free_callback (NULL);
   external_string = jerry_create_external_string ((jerry_char_t *) external_1, NULL);
   TEST_ASSERT (free_count == 1);
   jerry_release_value (external_string);
   TEST_ASSERT (free_count == 1);
 
-  external_string = jerry_create_external_string ((jerry_char_t *) external_2, free_external2);
+  jerry_string_set_external_string_free_callback (external_string_free_callback_2);
+  external_string = jerry_create_external_string ((jerry_char_t *) external_2, (void *) &free_count);
   TEST_ASSERT (free_count == 2);
   jerry_release_value (external_string);
   TEST_ASSERT (free_count == 2);
```

Comment:
Shouldn't the `free_count` change after the string release? Or am I missing something? (Maybe we need a force gc() call?)

Reply:
Nope. external_2 is `"Object"`, and a magic string is used, so the callback is called by `jerry_create_external_string`.

Reply:
This is always worked this way. See `ecma_new_ecma_external_string_from_cesu8`. I will add some words about it in the documentation, that is a good idea.